benchmarks/test_differentiation.py

```diff
@@ -0,0 +1,77 @@
+"""Benchmarks for differentiation using ProtoSym and SimpleCAS.
+
+Differentiation using ProtoSym and SimpleCAS is currently benchmarked against
+SymPy's and SymEngine's symbolic differentiation.
+
+"""
+from typing import Callable
+from typing import TypeVar
+
+import pytest
+import symengine
+import sympy
+
+from protosym import simplecas
+
+
+ExprType = TypeVar("ExprType")
+Fixture = Callable[..., ExprType]
+
+
+@pytest.mark.benchmark(group="differentiate nested sine first derivative")
+class TestNestedSineFirstDerivative:
+    """Differentiate ``sin(sin(sin(sin(sin(sin(x))))))`` w.r.t. ``x`` once."""
+
+    @staticmethod
+    def test_simplecas(benchmark: Fixture[simplecas.Expr]) -> None:
+        """Differentiate using SimpleCAS."""
+        x = simplecas.Symbol("x")
+        sin = simplecas.sin
+        expr = sin(sin(sin(sin(sin(sin(x))))))
+        benchmark(expr.diff, x)
```

Comment:
There should be some check that the output is correct like:
```python
result = benchmark(expr.diff, x)
assert result.eval_f64({x: 1.0}) == ...
```

Reply:
The benchmark result should never be shown if the result returned was not checked for correctness in some way.

Reply:
Valuable point, thanks. Addressed in https://github.com/oscarbenjamin/protosym/pull/40/commits/ddec9f7404520a425bbc23f01d795027b2ad7a13.